src/main/java/org/terasology/biomesAPI/BaseConditionalBiome.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.biomesAPI;
+
+import org.terasology.math.geom.Vector2f;
+import org.terasology.world.generation.facets.base.FieldFacet2D;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public abstract class BaseConditionalBiome implements ConditionalBiome {
+    protected Map<Class<? extends FieldFacet2D>, Vector2f> limitedFacets = new HashMap<>();
```

Comment:
Instead of using `Vector2f` to store the range, maybe consider using the [`Range` class from Guava?](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/Range.html). Handily, it would come with `range.contains(value)` for even simpler checks 😉